Marlin/src/lcd/ultralcd.h

```diff
@@ -292,26 +292,28 @@ class MarlinUI {
     static void resume_print();
 
     #if HAS_PRINT_PROGRESS
+      #if HAS_PRINT_PROGRESS_PERMYRIAD
+        typedef uint16_t progress_t;
+        #define PROGRESS_SCALE 100
+        #define PROGRESS_MASK 0x7FFF
+      #else
+        typedef uint8_t progress_t;
+        #define PROGRESS_SCALE 1
+        #define PROGRESS_MASK 0x7F
+      #endif
       #if ENABLED(LCD_SET_PROGRESS_MANUALLY)
-        #if HAS_PRINT_PROGRESS_PERMYRIAD
-          static uint16_t progress_override;
-          static void set_progress_permyriad(const uint16_t progress) { progress_override = _MIN(progress, 10000); }
-          static void set_progress(const uint8_t progress) { set_progress_permyriad(progress * 100); }
-          static void set_progress_done() { progress_override = 0x8000 + 10000; }
-          static void progress_reset() { if (progress_override & 0x8000) set_progress_permyriad(0); }
-        #elif HAS_PRINT_PROGRESS_PERCENT
-          static uint8_t progress_override;
-          static void set_progress(const uint8_t progress) { progress_override = _MIN(progress, 100); }
-          static void set_progress_done() { progress_override = 0x80 + 100; }
-          static void progress_reset() { if (progress_override & 0x80) set_progress(0); }
-        #endif
+        static progress_t progress_override;
+        static void set_progress(const uint8_t p) { progress_override = _MIN(p, 100) * PROGRESS_SCALE; }
```

Comment:
I want to have ability to set fractional percents via M73 Pxx.xx, as this done in klipper. So for this reason I included specific set_progress_permyriad() with uint16

Reply:
Now made with single set_progress and PROGRESS_SCALE, as set_progress used only by M73

Reply:
I am preserving that, as can be seen in `M73`. I am also preserving this.

Reply:
Ok, sounds good.